flink-connector-cassandra/src/test/java/org/apache/flink/connector/cassandra/source/CassandraTestEnvironment.java

```diff
@@ -86,25 +80,50 @@ public class CassandraTestEnvironment implements TestResource {
                     + KEYSPACE
                     + " WITH replication= {'class':'SimpleStrategy', 'replication_factor':1};";
 
-    // official Cassandra image deactivates jmx, to enable it we need to provide authentication and
-    // modify cassandra-env.sh so we created our own image
-    private static final String DOCKER_CASSANDRA_IMAGE_WITH_JMX = "echauchot/cassandra-jmx:4.0.8";
-
     static final String SPLITS_TABLE = "flinksplits";
     private static final String CREATE_SPLITS_TABLE_QUERY =
             "CREATE TABLE " + KEYSPACE + "." + SPLITS_TABLE + " (id int PRIMARY KEY, counter int);";
     private static final String INSERT_INTO_FLINK_SPLITS =
             "INSERT INTO " + KEYSPACE + "." + SPLITS_TABLE + " (id, counter)" + " VALUES (%d, %d)";
     private static final int NB_SPLITS_RECORDS = 1000;
 
-    @Container private final CassandraContainer cassandraContainer = createCassandraContainer();
-
-    @TempDir private File tempDir;
+    @Container
+    private final CassandraContainer cassandraContainer;
+    private final String dockerHostIp;
+    private final int jmxPort;
 
-    private static Cluster cluster;
-    private static Session session;
+    private Cluster cluster;
+    private Session session;
     private ClusterBuilder clusterBuilder;
 
+
+    public CassandraTestEnvironment() {
+        try (ServerSocket s = new ServerSocket(0)) {
+            // use fixed free random port for JMX
+            jmxPort = s.getLocalPort();
+            // resolve IP of docker host to establish JMX connection
+            dockerHostIp = InetAddress.getLocalHost().getHostAddress();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        cassandraContainer =
+                new CassandraContainer("cassandra:4.0.8");
+        // use a fixed port mapping for JMX, it doesn't work well with mapped ports
+        cassandraContainer.setPortBindings(ImmutableList.of(jmxPort + ":" + jmxPort));
+        // more generous timeouts and remote JMX configuration
+        addJavaOpts(
+                cassandraContainer,
+                "-Dcassandra.request_timeout_in_ms=30000",
+                "-Dcassandra.read_request_timeout_in_ms=15000",
+                "-Dcassandra.write_request_timeout_in_ms=6000",
+                "-Dcassandra.jmx.remote.port=" + jmxPort,
+                "-Dcom.sun.management.jmxremote.rmi.port=" + jmxPort,
+                "-Djava.rmi.server.hostname=" + dockerHostIp,
+                "-Dcom.sun.management.jmxremote.host=" + dockerHostIp
+        );
+    }
+
```

Comment:
I tried similar thing using `System.setProperty("cassandra.jmx.....", value);` but it failed due to the fact that these system props are not propagated to the container my testContainers. Thanks for the hint to pass props properly